partiql-ast/src/main/java/org/partiql/ast/TableConstraint.java

```diff
@@ -0,0 +1,94 @@
+package org.partiql.ast;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import org.jetbrains.annotations.NotNull;
+import org.partiql.ast.expr.Expr;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public abstract class TableConstraint extends AstNode {
+
+    @Builder(builderClassName = "Builder")
+    @EqualsAndHashCode(callSuper = false)
+    public static class Unique extends TableConstraint {
+        // TODO: Should this be an identifier
```

Comment:
SQL BNF seems to use more than just a string name -- https://ronsavage.github.io/SQL/sql-99.bnf.html#constraint%20name.

Also `name` could be a field of `TableConstraint` rather than each of the variants.

Reply:
SQL has that as identifier since it requires constraint name to be unique within the same schema... 

The constraint name semantics is quite hand wavy now in PartiQL... 

Let's discuss a little more ...

Reply:
To followup on this comment, will the PartiQL RFC for CREATE TABLE opt for a string name here or an identifier?

Reply:
Actually: let's make the constraint name a identifier (simple identifier) for now...

Not sure what would end up being in the RFC. Practically: we probably just going to leave the semantics of identifier name out of scope. Something like: 

> The PartiQL parser supports parsing of constraint name (as simple identifier, regular or delimited). However, the operation semantics for constraint name is out of scope. 

In implementation, practically speaking, as PartiQL does not have a persistence metadata layer, it is difficult for it to support things like assertion constraint name conflict across schema, or operations such as drop constraint via name. (Potentially possible via SPI, but not in the road map as for today).

Reply:
Discussed offline: In the sprit of following SQL, will make the constraint name a IdentifierChain.